coalib/results/AbsolutePosition.py

```diff
@@ -0,0 +1,57 @@
+from coalib.results.TextPosition import TextPosition
+from coalib.misc.Decorators import enforce_signature
+
+
+class AbsolutePosition(TextPosition):
+
+    @enforce_signature
+    def __init__(self,
+                 text: (tuple, list, None)=None,
+                 position: (int, None)=None):
+        """
+        Creates an AbsolutePosition object that represents the index of a
+        character in a string.
+
+        :param text:     The text containing the character.
+        :param position: Position identifying the index of character
+                         in text.
+        """
+        line = column = None
+        if position and text:
+            line, column = calc_line_col(text, position)
+        self._text = text
+        self._position = position
+        super().__init__(line, column)
+
+    @property
+    def position(self):
+        return self._position
+
+
+def calc_line_col(text, position):
+    """
+    Creates a tuple containing (line, column) by calculating line number
+    and column in the text, from position. Line and column start from 1.
+    Example:
+
+    >>> text = ("a\\n",)
+
+    Now here we can see the index of 'a' in ''.join(text) is 0, so the
+    corresponding line and column of 'a' should be:
+
+    >>> calc_line_col(text, 0)
+    (1, 1)
```

Comment:
I find it very important to give the user an example with e.g. the first char of the second line.  Also the definition of `text` in an own line/example is a bit verbose IMO, can we do something like this (properly phrased, I'm just dumping raw thoughts as text here):

```
The position starts with zero as you can see:

>>> calc_line_col(('a\n',), 0)
(1, 1)

Now if we enter a second line, the newline character has it's own position
and counts into the number:

>>> calc_line_col(('a\n', 'b\n'), 1)
(1, 2)
>>> calc_line_col(('a\n', 'b\n'), 2)
(2, 1)
```

Also for all inline code samples like `''.join(text)` you should surround those by double backticks